### PR TITLE
CANDLEPIN-762: Update overrides of openapi-generator to match new package version

### PR DESCRIPTION
**Pull request by wottop**
This is a tricky one. The new version of the package treats the nullification of collections differently.
Both the spec client and the base Candlepin require null to be a value in certain situations.
This causes some issues where we were expecting pre-initialization.
While the tests now pass, we will need to confirm it with Katello to ensure that they are not surprised by what we return from a call.

### CODE REVIEW

File: src/main/java/org/candlepin/model/OwnerInfoBuilder.java [hunk 1]
```diff
@@ -32,6 +33,14 @@ public class OwnerInfoBuilder {
 
     public OwnerInfoBuilder() {
         ownerInfo = new OwnerInfo();
+        ownerInfo.setConsumerTypeCountByPool(new LinkedHashMap<>());
```
Comment (nikosmoum) [inline src/main/java/org/candlepin/model/OwnerInfoBuilder.java:36]: same question here

File: spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorCheckInSpecTest.java [hunk 2]
```diff
@@ -1613,10 +1613,15 @@ private HypervisorUpdateResultDTO getResultData(AsyncJobStatusDTO status) throws
 
     private void confirmResultDataCounts(HypervisorUpdateResultDTO resultData, int created, int updated,
         int unchanged, int failedUpdate) {
-        assertEquals(created, resultData.getCreated().size());
-        assertEquals(updated, resultData.getUpdated().size());
-        assertEquals(unchanged, resultData.getUnchanged().size());
-        assertEquals(failedUpdate, resultData.getFailedUpdate().size());
+        int createdCount = resultData.getCreated() == null ? 0 : resultData.getCreated().size();
```
[the remark this answers is not in the document]
Comment (nikosmoum) [reply spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorCheckInSpecTest.java:1616]: Or more to the point: what has the hypervisor result data behaviour been up until now? were these Sets always returned as empty arrays, or as null by default? Because this seems like the API changed, and we're changing the test to match, which does not look right

File: src/main/java/org/candlepin/model/OwnerInfoBuilder.java (hunk 1 above)
Comment (nikosmoum) [reply src/main/java/org/candlepin/model/OwnerInfoBuilder.java:36]: Looking back at candlepin-4.0-HOTFIX, empty maps has been the correct behaviour here, so all good

File: spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorCheckInSpecTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikosmoum) [reply spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorCheckInSpecTest.java:1616]: Looking back at old code, the server has always been returning these as empty arrays: https://github.com/candlepin/candlepin/blob/candlepin-2.3-HOTFIX/server/src/main/java/org/candlepin/resource/dto/HypervisorUpdateResult.java#L42-L46 so we need to make sure that continues, and this test should be reverted to always expect them as non-null, empty Sets

**Comment by wottop** [comment]
I left these as separate commits. Let me know if it needs a squash.

**Review by nikosmoum — APPROVED** [review]
